test/unit/mongo/MongoSearchProviderTest.php

```diff
@@ -178,15 +183,26 @@ public function testSearchIndexingRemovesDocWhenTypeHasNoCorrespondingSearchdocS
     {
         // update a document
         $id = array('_id.r'=>'http://talisaspire.com/resources/doc1');
-        $this->tripod->collection->update($id, array('$set'=>array("rdf:type"=>array("u"=>"bibo:Article"))));
+        $this->getTripodCollection($this->tripod)
+            ->update($id, array('$set'=>array("rdf:type"=>array("u"=>"bibo:Article"))));
 
         // reindex
-        $this->indexer->generateAndIndexSearchDocuments('http://talisaspire.com/resources/doc1', 'http://talisaspire.com/', $this->tripod->getCollectionName());
+        $this->indexer->generateAndIndexSearchDocuments('http://talisaspire.com/resources/doc1', 'http://talisaspire.com/', $this->tripod->getPodName());
+
+        $actualSearchDocumentCount = $this->getCountForSearchSpecs($this->tripod);
 
-        $actualSearchDocumentCount = $this->tripod->db->selectCollection(SEARCH_INDEX_COLLECTION)->count(array());
         $this->assertEquals(11, $actualSearchDocumentCount, "Should only be 11 search documents now that one of them has had its type changed with no corresponding search doc spec");
 
-        $result = $this->tripod->db->selectCollection(SEARCH_INDEX_COLLECTION)->findOne(array("_id.r"=>"http://talisaspire.com/resources/doc1"));
+
+        foreach(MongoTripodConfig::getInstance()->getCollectionsForSearch('tripod_php_testing') as $collection)
+        {
+            $result = $collection->findOne(array("_id.r"=>"http://talisaspire.com/resources/doc1"));
+            if($result)
```

Comment:
Is this just making sure there are no results in any of the search collections? If it finds a result, break out, then assert $result is null? You could put the assertNull inside the foreach and make sure every result is null rather than breaking out?